### ISSUE TITLE
Is it possible to add bind method to iOS?

### ISSUE THREAD

**Issue by DmyMi** [icerockdev/moko-media#44]
Hello, the Android implementation (for example, `MediaPickerController`) has the `bind(lifecycle: Lifecycle, fragmentManager: FragmentManager)` method that has one other advantage except lifecycle management - it is common DI friendly, i.e. i can inject the picker into ViewModel and then bind it when ever i want.

But for iOS side you need to pass the `UIViewController` to the constructor of `MediaPickerController` which makes it hard to inject it in common code with, for example, `Kodein`. To make it work i created a copy of your `MediaPickerController` with the bind/unbind methods on iOS side and it works. Is it possible for you to implement something similar for people using common DI?

Example use case i have for it:

* ios picker with bind

```kotlin
actual class ImagePicker(
    val permissionsController: PermissionsController // ,
//    private val getViewController: () -> UIViewController
) {
//    constructor(
//        permissionsController: PermissionsController,
//        viewController: UIViewController
//    ) : this(
//        permissionsController = PermissionsController,
//        getViewController = { viewController }
//    )

    private var getViewController: (() -> UIViewController)? = null

    fun bind(viewController: UIViewController) {
        getViewController = { viewController }
    }

    fun unbind() {
        getViewController = null
    }
// other methods
}
```

* view model using it

```kotlin
class AccountViewModel constructor(
    // other dependencies
    val imagePicker: ImagePicker
) : BaseViewModel() {}
```

* kodein config

```kotlin
// copied relevant code from different modules
    bind() from provider {
        PermissionsController()
    }
//...
    bind() from provider {
        ImagePicker(instance())
    }
//...
    bind() from provider {
        AccountViewModel(instance() /* the picker */, /* ... other dependencies */)
    }
//...
    fun getAccountViewModel() = kodein.direct.instance<AccountViewModel>()
```

* SwiftUI ViewModel wrapper

```swift
class AccountViewModelWrap: BaseViewModelWrap {
    private let vm: AccountViewModel = InjectLocator().getAccountViewModel()

    func bindViewController(viewController: UIViewController) {
        self.vm.imagePicker.bind(viewController: viewController)
    }
// other code
```

* SwiftUi view

```swift
struct AccountView: View {
    @StateObject var viewModel = AccountViewModelWrap()
    
    var body: some View {
        GeometryReader { geometry in
            NavigationView {
                ZStack {
                    PickerViewController(viewModel: self.viewModel)
                    VStack {
                   // other views
                   }
               }
            }
        }
    }
```

* finally, picker view controller representable to bind the `UIViewController`

```swift
struct PickerViewController : UIViewControllerRepresentable {
    
    @ObservedObject var viewModel: AccountViewModelWrap
    
    func makeUIViewController(context: Context) -> UIViewController {
        let ctrl = UIViewController()
        self.viewModel.bindViewController(viewController: ctrl)
        return ctrl
    }
    
    // other stuff
}
```

and unbind is called onDisappear of the parent view.

Or maybe if you have a simpler solution - can you show please? :)

**Comment by Alex009** [comment]
yes, it can be implemented. we should save backward compatibility too, but this idea will be implemented

**Comment by Alex009** [comment]
after small research i think we should implement better support of SwiftUI than just bind method. as i see in samples https://www.hackingwithswift.com/books/ios-swiftui/importing-an-image-into-swiftui-using-phpickerviewcontroller we should give our picker viewcontroller to swiftui.

**Comment by Alex009** [comment]
will be released in 0.11.0
